### PR TITLE
[stable/redis] add init-container for set chown redis data dir

### PR DESCRIPTION
**Pull request by LuckySB**
#### What this PR does / why we need it:
with .Values.master.securityContext.enabled Redis run as non privileged user, but some persistence volume provisioners (such as cephfs) mount volume with owner root, so redis cannot create backup files - appendonly.aof or dump.rdb, and crashed

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [ x] Chart Version bumped
- [ x] Variables are documented in the README.md

### CODE REVIEW

**Comment by juan131** [comment]
Hi @LuckySB 

According to the docs:
https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#discussion

> Volumes that support ownership management are modified to be owned and writable by the GID specified in fsGroup.

We're already setting a **securityContext** on the Redis Statefulset. Therefore, I wonder if we can give writing permissions to the group instead of changing the ownership. Sth like:

```
command: ['sh', '-c', 'chmod -R g+rwX {{ .Values.master.persistence.path }}']
```

**Comment by juan131** [comment]
🤔 you're right... The init container does not make any harm anyway.. Let's proceed with your approach. Thanks for bringing this up!

File: stable/redis/values-production.yaml [hunk 1]
```diff
@@ -317,6 +317,18 @@ metrics:
...
 
+##
+## Init containers parameters:
+## volumePermissions: Change the owner of the persist volume mountpoint to RunAsUser:fsGroup
+##
+
+volumePermissions:
+  image:
+    registry: docker.io
+    repository: alpine
```
Comment (juan131) [inline stable/redis/values-production.yaml]: Can we use `busybox` (1.2.9) instead? It's about half the size of alpine...

https://hub.docker.com/r/library/busybox/tags/
https://hub.docker.com/r/library/alpine/tags/

File: stable/redis/Chart.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: redis
-version: 4.0.1
+version: 4.0.2
```
Comment (juan131) [inline stable/redis/Chart.yaml]: Increase minor version instead.

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/redis/values-production.yaml (hunk 1 above)
Comment (LuckySB) [reply stable/redis/values-production.yaml]: ok.

File: stable/redis/Chart.yaml (hunk 2 above)
Comment (LuckySB) [reply stable/redis/Chart.yaml]: ok

**Review by juan131 — APPROVED** [review]
Great thanks for using busybox!

**Comment by desaintmartin** [comment]
Thanks for this.
@ maintainers, @juan131 , would it be wise to configure host sysctl settings with this initContainer? I get
```
WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
28:M 02 Aug 12:13:31.423 # Server initialized
28:M 02 Aug 12:13:31.423 # WARNING you have Transparent Huge Pages (THP) support enabled in your kernel. This will create latency and memory usage issues with Redis. To fix this issue run the command 'echo never > /sys/kernel/mm/transparent_hugepage/enabled' as root, and add it to your /etc/rc.local in order to retain the setting after a reboot. Redis must be restarted after THP is disabled.
```
I know elasticsearch chart does it.
If so, I volunteer to do it.

**Pull request merged**